openAPI/user-related-apis/v1.json

```diff
@@ -1 +1 @@
boolean","example":false}},"additionalProperties":false}}}},"504":{"description":"The server did not receive a timely response from the upstream server.","content":{"application/json":{"schema":{"type":"object","properties":{"message":{"type":"string","example":"The server was acting as a gateway or proxy and did not receive a timely response from the upstream server"},"type":{"type":"string","example":"gateway_timeout"},"success":{"type":"boolean","example":false}},"additionalProperties":false}}}}}},"put":{"description":"Update a post.","tags":["Posts"],"summary":"Update a post","parameters":[{"in":"path","name":"postId","schema":{"type":"string","description":"The id of the comment to be updated.","example":1234},"required":true,"description":"The id of the comment to be updated."}],"requestBody":{"content":{"application/json":{"schema":{"type":"object","properties":{"body":{"type":"string","description":"The body of the post.","example":"I love Allah and his messenger."},"roomId":{"type":"integer","description":"The Id of the room the post will be posted in.","example":123},"isPrivate":{"type":"boolean","description":"Whether the post should be posted as a private note.","example":false,"default":false},"tagIds":{"type":"array","items":{"type":"integer"},"description":"The Ids of the tags the post is attached to","example":[432]},"references":{"type":"array","items":{"$ref":"#/components/schemas/PostReference"},"description":"The Surahs/Ayahs the post relates to. If `references` is not passed with update post API, existing references of the post will be removed.","example":[{"surahId":2},{"surahId":5,"fromAyah":6,"toAyah":8},{"surahId":114,"fromAyah":2,"toAyah":2}]}},"required":["body"],"additionalProperties":false}}}},"responses":{"200":{"description":"Request has been handled successfully.","content":{"application/json":{"schema":{"type":"object","properties":{"success":{"type":"boolean","example":true},"data":{"type":"object","properties":{"message":{"enum":["Post was updated successfully"]}},"additionalProperties":false}},"additionalProperties":false}}}},"400":{"description":"The request is missing required parameters or is invalid.","content":{"application/json":{"schema":{"type":"object","properties":{"message":{"type":"string","example":"The request is missing required headers or is invalid"},"type":{"type":"string","example":"invalid_request"},"success":{"type":"boolean","example":false}},"additionalProperties":false}}}},"401":{"description":"The request is unauthorized.","content":{"application/json":{"schema":{"type":"object","properties":{"message":{"type":"string","example":"The request requires user authentication"},"type":{"type":"string","example":"unauthorized"},"success":{"type":"boolean","example":false}},"additionalProperties":false}}}},"403":{"description":"Forbidden error. Can either be due to access token not being passed, having been expired or the caller trying to access a resource without enough permissions.","content"
```

Comment:
JAK for the changes Basit, however, there is an issue with changing this file which is that it will always be overriden with every new release of our auth backend because we have this CI/CD pipeline that takes the schema from the code itself, generate the file then push it to this repo and generate the documentation afterwards. For any user-related APIs, the change needs to come from the code of the auth backend itself. inshAllah I will work on that myself. so I suggest we revert everything related to user APIs and keep only v4 APIs.